packages/InteractiveIcon/IconButton.jsx

```diff
@@ -0,0 +1,133 @@
+import React, { forwardRef } from 'react'
+import PropTypes from 'prop-types'
+import styled, { ThemeProvider } from 'styled-components'
+
+import A11yContent from '@tds/core-a11y-content'
+import { colorGreyGainsboro, colorGreyShark, colorTelusPurple, colorWhite } from '@tds/core-colours'
+import { safeRest } from '@tds/util-helpers'
+import { componentWithName } from '@tds/util-prop-types'
+
+import animations from './shared/animations'
+import SharedStyledInteractiveIconButton from './shared/StyledInteractiveIconButton'
+import SharedStyledInteractiveIconHover from './shared/StyledInteractiveIconHover'
+import { warn } from '../../shared/utils/warn'
+
+export const StyledIconButton = styled(SharedStyledInteractiveIconButton)(
+  animations.scale,
+  animations.reduceMotion,
+  ({ variant }) => {
+    // add colour to the DependentIcon
+    if (variant === 'alternative') {
+      return {
+        '& svg': {
+          fill: colorTelusPurple,
+        },
+      }
+    }
+    if (variant === 'inverted') {
+      return {
+        '& svg': {
+          fill: colorWhite,
+        },
+      }
+    }
+    return {
+      '& svg': {
+        fill: colorGreyShark,
+      },
+    }
+  }
+)
+
+const getTheme = variant => {
+  if (variant === 'alternative') {
+    return {
+      hoverBackgroundColor: '#D8CBE5',
+      iconColor: colorTelusPurple,
```

Comment:
`iconColor` is a variable used as part of our `ThemeProvider` context. `StyledInteractiveIconSVG` consumes the theme and uses `iconColor` as `fill`.

Reply:
I thought we were using Dependent icons' `color` prop instead of `fill` CSS. Let's remove the use of `fill` in StyledInteractiveIconSVG` and use `color` prop instead.